indigo_social/views.py

```diff
@@ -292,3 +293,17 @@ class UserPopupView(AbstractAuthedIndigoView, DetailView):
     slug_url_kwarg = 'username'
     template_name = 'indigo_social/user_popup.html'
     queryset = User.objects
+
+
+class UserProfilePhotoView(View):
```

Comment:
```suggestion
class UserProfilePhotoView(AbstractAuthedIndigoView):
```

That view will ensure a user is logged in if the site requires it.